telemetry/telemetry_record.py

```diff
@@ -73,15 +100,48 @@ def move_wpr_files(name_schema):
     """
     remote_data_path = os.path.join(CHROMIUM_SRC, 'tools/perf/page_sets/data/',
             name_schema)
-    local_path = os.path.join(os.getcwd(), 'data/')
+    local_path = os.path.join(os.getcwd(), 'data/wpr_source')
     p = Popen('mv {0} {1}'.format(remote_data_path, local_path), shell=True)
 
+def make_modified_json(num_urls):
+    """Creates *.json files to point to the right archive files
+
+    Located in ~/src/tools/perf/page_sets/data
+    :param num_urls: int for the number of target urls
+    TODO: add sha1?
+    """
+    json_path = '/home/jamshed/src/tools/perf/page_sets/data/'
+    json_template = ('{{'
+            '"description": "Describes the Web Page Replay archives for a user '
+            'story set. Dont edit by hand! Use record_wpr for updating.", '
+            '"archives": {{ "url{0}_page_set_000.wpr": ["url{0}"]}}}}\n')
+    for i in range(num_urls):
+        modified_name = '999999' + str(i)
+        file_name = 'url{0}_page_set.json'
+        with open('{0}{1}'.format(json_path,
+            file_name.format(modified_name)), 'wb') as f:
+            f.write(json_template.format(modified_name))
+
 def modify_wpr():
-    """Sets cacheable objects delay time to 0, creates  pc.wpr files"""
-    wpr_directory = os.path.join(os.getcwd(), 'data/')
+    """Sets cacheable objects delay time to 0, creates pc.wpr files
+
+    Note: *.pc.wpr files are converted to 999999 + str(url name), which lets
+    them be treated like regular urls. Just know that six 9's indicate a
+    modified wpr file
+    TODO: Fix the 999999 notation
```

Comment:
Right, that's what I mean: getting the result from each benchmark
in-memory, so that we don't need to care about the name of the benchmark
python class. Is that problematic?

Disk is fine for IPC as long as you don't care about performance, which we
don't really. At least, adb is a far bigger perf issue than this.

On Fri, Apr 17, 2015 at 10:24 PM, Jamshed Vesuna <notifications@github.com>
wrote:

> In telemetry/telemetry_record.py
> <https://github.com/colin-scott/page_load_time/pull/2#discussion_r28641565>
> :
>
> >  def modify_wpr():
> > -    """Sets cacheable objects delay time to 0, creates  pc.wpr files"""
> > -    wpr_directory = os.path.join(os.getcwd(), 'data/')
> > +    """Sets cacheable objects delay time to 0, creates pc.wpr files
> > +
> > +    Note: *.pc.wpr files are converted to 999999 + str(url name), which lets
> > +    them be treated like regular urls. Just know that six 9's indicate a
> > +    modified wpr file
> > +    TODO: Fix the 999999 notation
>
> Right - the benchmark class names do not need the url in their name. They
> do, however, need some sort of unique identifier for the url (otherwise
> there's no way to tell each benchmark apart)? Unless, what you mean is that
> after running a single benchmark for a single url, we check the output,
> read that into memory, and continue. Is this what you mean?
>
> Also, I use the disk as a way for two processes to communicate. The
> benchmark has some results that need to get passed into this program. Any
> thoughts on how to better communicate this data?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/colin-scott/page_load_time/pull/2/files#r28641565>.
>